### DISCUSSION TITLE
load user, his teams, assigned users with role name

### DISCUSSION THREAD

**Discussion by VictoriaNici** [spatie/laravel-permission#2502] [Q&A]
Why this statement not loading user roles for user teams,
at Services/Users/GetUserProfile.php:
`$this->user->load(['groups.users.roles']);`
I am getting groups(teams), assigned users(belongsToMany relation) but without their roles. Trying different workarounds, nothing works.
also I have this in User model. How can I properly integrate the roleName method within the load statement to retrieve user roles?
at Models/User.php:
```
public function roleName(string $teamId, string $scope): ?string
    {
        $actualTeamUuid = getPermissionsTeamId();
        setPermissionsTeamId($teamId);
        $role = $this->roles->where(UserRole::COL_SCOPE, $scope)->first();

        setPermissionsTeamId($actualTeamUuid);

        return $role ? $role->name : null;
    }
```
    
It works in  foreaches, but how get the roles in load (it also not works, how much I tried). 
I want to mention that I applied some custom solutions (use uuids).
Thank you in advance for your help.

**Comment by erikn69** [comment]
https://github.com/spatie/laravel-permission/discussions/2483#discussioncomment-7048141
You could make your own custom query
